drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -1119,3 +1120,61 @@ def workbench_scenarios():
                 "<vertical_demo><drag-and-drop-v2 mode='assessment' max_attempts='3'/></vertical_demo>"
             ),
         ]
+
+    def _strip_html_tags_and_extra_whitespaces(self, raw_html):
+        cleanr = re.compile('<.*?>')
+        cleantext = str(re.sub(cleanr, '', raw_html))
+        cleantext = " ".join(re.split(r"\s+", cleantext, flags=re.UNICODE)).strip()
+        return cleantext
+
+    def index_dictionary(self):
+        """
+        Return dictionary prepared with module content and type for indexing.
+        """
+        # return key/value fields in a Python dict object
+        # values may be numeric / string or dict
+        # default implementation is an empty dict
+        xblock_body = super(DragAndDropBlock, self).index_dictionary()
+
+        zones_display_names = {
+            "zone_{}_display_name".format(zone_i):
+            self._strip_html_tags_and_extra_whitespaces(zone.get("title", ""))
+            for zone_i, zone in enumerate(self.data.get("zones", []))
+        }
+
+        zones_description = {
+            "zone_{}_description".format(zone_i):
+            self._strip_html_tags_and_extra_whitespaces(zone.get("description", ""))
+            for zone_i, zone in enumerate(self.data.get("zones", []))
+        }
+
+        items_display_names = {
+            "item_{}_display_name".format(item_i):
+            self._strip_html_tags_and_extra_whitespaces(item.get("displayName", ""))
+            for item_i, item in enumerate(self.data.get("items", []))
+        }
+
+        items_image_description = {
+            "item_{}_image_description".format(item_i):
+            self._strip_html_tags_and_extra_whitespaces(item.get("imageDescription", ""))
+            for item_i, item in enumerate(self.data.get("items", []))
+        }
+
+        index_body = {
+            "display_name": self.display_name,
+            "question_text": self._strip_html_tags_and_extra_whitespaces(self.question_text),
+            "background_image_description": self.data.get("targetImgDescription", ""),
+            **items_display_names,
+            **items_image_description,
+            **zones_display_names,
+            **zones_description
+        }
```

Comment:
Actually, could you go back to using the `.update` syntax? This might be more concise, but it breaks on Python 2.

Reply:
ofcourse